src/EFTools/EntityDesignerVersioningFacade/ReverseEngineerDb/SchemaDiscovery/EntityStoreSchemaGeneratorDatabaseSchemaLoader.cs

```diff
@@ -15,17 +18,44 @@ namespace Microsoft.Data.Entity.Design.VersioningFacade.ReverseEngineerDb.Schema
     /// </summary>
     internal class EntityStoreSchemaGeneratorDatabaseSchemaLoader
     {
+        private const string SwitchOffMetadataMergeJoins = "Switch.Microsoft.Data.Entity.Design.DoNotUseSqlServerMetadataMergeJoins";
         private readonly EntityConnection _connection;
         private readonly Version _storeSchemaModelVersion;
+        private readonly IDbCommandInterceptor _addOptionMergeJoinInterceptor;
 
-        public EntityStoreSchemaGeneratorDatabaseSchemaLoader(EntityConnection entityConnection, Version storeSchemaModelVersion)
+        [SuppressMessage("Microsoft.Usage", "CA1806:DoNotIgnoreMethodResults", MessageId = "System.Boolean.TryParse(System.String,System.Boolean@)")]
+        public EntityStoreSchemaGeneratorDatabaseSchemaLoader(
+            EntityConnection entityConnection,
+            Version storeSchemaModelVersion,
+            bool needsMergeJoinHint)
```

Comment:
I don't think we need to lift the decision to the ModelGenerator and add this parameter. Another alternative to the ones you mentioned before is to obtain the invariant name from `_connection.StoreProviderFactory.GetProviderInvariantName()`, so you actually already have the information you need in this class.

Reply:
Fixed. That's much better. (Btw it's actually `_connection.StoreConnection.GetProviderInvariantName()`).